plugins/ftdidmx/FtdiDmxThread.cpp

```diff
@@ -42,7 +51,15 @@ FtdiDmxThread::FtdiDmxThread(FtdiInterface *interface, unsigned int frequency)
   : m_granularity(UNKNOWN),
     m_interface(interface),
     m_term(false),
-    m_frequency(frequency) {
+    m_frequency(frequency),
+    m_transaction_number(0),
+    m_discovery_agent(this),
+    m_uid(0x7a70, 0x12345678),
```

Comment:
Can we add a todo for this at least!

Reply:
Totally forgot that, didn't you say there was a built-in UID I should use?

Reply:
The server has a UID, which is probably the best bet, check the plugin adaptor docs.

Reply:
I have not be able to find this UID, I'll try to change the logic to leverage the FTDI serial number for the device part of the UID while maintaining the Vendor part as is (like in a lot of other locations in the code, it may need to be defined in some RDM related enum).

Reply:
Changed to being based on the serial of the FTDI chip.

Whether UID should be set centrally for all interfaces and how is still under discussion.

Reply:
Sorry @Keeper-of-the-Keys , I finally found it, https://github.com/OpenLightingProject/ola/blob/cb021b41860ab63a5cb3add27e7fa91ac32511bb/olad/OlaServer.cpp#L188-L190 although it doesn't seem to be exposed in APIs currently. It does indeed default to the IP when one is present, or zero if there isn't one. See also this if we do them centrally https://github.com/OpenLightingProject/ola/issues/1344 .

Reply:
m_default_uid would need to be available through some getter function....
No reason to "calculate" a UID again for each interface.